chainermn/extensions/multi_node_evaluator.py

```diff
@@ -26,6 +29,23 @@ def create_multi_node_evaluator(actual_evaluator, communicator):
 
     def new_evaluate(self):
         local_mean_dict = self._mn_original_evaluate()
+
+        # ChainerX support:
+        # We need convert chainerx ndarray to Native array because
+        #   (1) allreduce_obj is used to compute global mean values
+        #   (2) allreduce_obj calls mpi4py.allreduce, which pickles the object
+        #   (3) chainerx.ndarray preserves CUDA device internally when pickled
+        #   (4) Error is caused when an ndarray is unpickled in another process
+        array0 = list(local_mean_dict.values())[0]
+        xp = chainer.backend.get_array_module(array0)
+        if xp == chx and array0.device.backend.name == 'cuda':
+            # Results of evaluation is fairly small, so
+            # the ndarray is transferred to CPU and allreduce()-ed.
```

Comment:
We can leave a comment here as there is a potential performance issue

Reply:
Fixed! Thanks